runtime/runtime_controller.cc

```diff
@@ -368,11 +396,20 @@ bool RuntimeController::HasLivePorts() {
 }
 
 tonic::DartErrorHandleType RuntimeController::GetLastError() {
-  std::shared_ptr<DartIsolate> root_isolate = root_isolate_.lock();
+  std::shared_ptr<DartIsolate> root_isolate = GetRootIsolate().lock();
   return root_isolate ? root_isolate->GetLastError() : tonic::kNoError;
 }
 
 std::weak_ptr<DartIsolate> RuntimeController::GetRootIsolate() {
+  std::shared_ptr<DartIsolate> root_isolate = root_isolate_.lock();
+  if (root_isolate) {
+    return root_isolate_;
+  }
+
+  // Root isolate is not yet created, get it and do some configuration.
+  FML_DCHECK(create_and_config_root_isolate_.valid());
+  create_and_config_root_isolate_.get();
```

Comment:
To avoid race conditions, should we assert here that `GetRootIsolate` must be called in a single thread? Otherwise, `create_and_config_root_isolate_.valid()` could become `false` right before `create_and_config_root_isolate_.valid()`.

Reply:
As the comments of class `RuntimeController`, "This object may only be created, used, and collected on the UI
 task runner.", Any codes access it in other thread is breaking the design and should be corrected. So there should be no race conditions.